File: Database/Assembly_SRPG_JP/TobiraCondsParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.TobiraCondsParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

using System.Collections.Generic;

namespace SRPG
{
  public class TobiraCondsParam
  {
    private List<TobiraConditionParam> mConditions = new List<TobiraConditionParam>();
    private string mUnitIname;
    private TobiraParam.Category mCategory;

    public string UnitIname
    {
      get
      {
        return this.mUnitIname;
      }
    }

    public TobiraParam.Category TobiraCategory
    {
      get
      {
        return this.mCategory;
      }
    }

    public TobiraConditionParam[] Conditions
    {
      get
      {
        return this.mConditions.ToArray();
      }
    }

    public void Deserialize(JSON_TobiraCondsParam json)
    {
      if (json == null)
        return;
      this.mUnitIname = json.unit_iname;
      this.mCategory = (TobiraParam.Category) json.category;
      this.mConditions.Clear();
      if (json.conds == null)
        return;
      for (int index = 0; index < json.conds.Length; ++index)
      {
        TobiraConditionParam tobiraConditionParam = new TobiraConditionParam();
        tobiraConditionParam.Deserialize(json.conds[index]);
        this.mConditions.Add(tobiraConditionParam);
      }
    }
  }
}

File: Database_V2/ParamFunctions_untested/TobiraCondsParam.py
def TobiraCondsParam(json):
    this={}#TobiraCondsParamjson)
    #if(json==null)
    #return
    if 'unit_iname' in json:
        this['mUnitIname'] = json['unit_iname']
    if 'category' in json:
        this['mCategory'] = json['category']
    #this.mConditions.Clear()
    #if(json.conds==null)
    #return
    #for(intindex=0index<json.conds.Length++index)
        #TobiraConditionParamtobiraConditionParam=newTobiraConditionParam()
        #tobiraConditionParam.Deserialize(json.conds)
        #this.mConditions.Add(tobiraConditionParam)
return this
